common/storage/sql/sqlite3/storage.c

```diff
@@ -357,3 +371,168 @@ retcode_t iota_stor_transaction_update(const connection_t* const conn,
+                         NULL);
+  if (rc != SQLITE_OK) {
+    log_error(SQLITE3_LOGGER_ID,
+              "Failed in binding, sqlite3 code is: %\" PRIu64 \"\n", rc);
+    return RC_SQLITE3_FAILED_BINDING;
+  }
+  rc = sqlite3_step(sqlite_statement);
+  if (rc != SQLITE_OK && rc != SQLITE_DONE) {
+    rc = sqlite3_finalize(
+        sqlite_statement);  //  Finalize the prepared statement.
+    log_error(SQLITE3_LOGGER_ID,
+              "Failed in step, sqlite3 code is: %\" PRIu64 \"", rc);
+    return RC_SQLITE3_FAILED_STEP;
+  }
+
+  rc = sqlite3_finalize(sqlite_statement);  //  Finalize the prepared statement.
+  if (rc != SQLITE_OK) {
+    log_error(SQLITE3_LOGGER_ID,
+              "Failed finalizing statement, sqlite3 code is: %\" PRIu64 \"",
+              rc);
+    return RC_SQLITE3_FAILED_FINALIZE;
+  }
+
+  return RC_OK;
+}
+
+retcode_t iota_stor_milestone_load(const connection_t* const conn,
+                                   const char* col_name, const trit_array_p key,
+                                   iota_milestones_pack* pack) {
+  assert(col_name && strcmp(col_name, ""));
+  retcode_t ret = RC_OK;
+  char statement[MILESTONE_MAX_SELECT_STATEMENT_SIZE];
+  char const* err_msg = 0;
+  sqlite3_stmt* sqlite_statement = 0;
+
+  if ((ret = iota_milestone_select_statement(
+           col_name, statement, MILESTONE_MAX_SELECT_STATEMENT_SIZE))) {
+    return ret;
+  }
+
+  pack->insufficient_capacity = false;
+
+  int rc = sqlite3_prepare_v2((sqlite3*)conn->db, statement, -1,
+                              &sqlite_statement, &err_msg);
+  if (rc != SQLITE_OK) {
+    sqlite3_finalize(sqlite_statement);
+    return RC_SQLITE3_FAILED_PREPARED_STATEMENT;
+  }
+
+  rc = sqlite3_bind_blob(sqlite_statement, 1, (void*)key->trits, key->num_bytes,
+                         NULL);
+
+  if (rc != SQLITE_OK) {
+    sqlite3_finalize(sqlite_statement);  //  Finalize the prepared statement.
+    log_error(SQLITE3_LOGGER_ID,
+              "Failed in binding, sqlite3 code is: %\" PRIu64 \"", rc);
+    return RC_SQLITE3_FAILED_BINDING;
+  }
+
+  pack->insufficient_capacity = false;
+  while (sqlite3_step(sqlite_statement) ==
+         SQLITE_ROW) {  // While query has result-rows.
+    if (pack->num_loaded == pack->txs_capacity) {
+      pack->insufficient_capacity = true;
+      break;
+    }
+    select_milestones_populate_from_row(sqlite_statement,
+                                        pack->milestones[pack->num_loaded++]);
+  }
+
+  rc = sqlite3_finalize(sqlite_statement);  //  Finalize the prepared statement
+  if (rc != SQLITE_OK) {
+    log_error(SQLITE3_LOGGER_ID,
+              "Failed in finalizing, sqlite3 code is: %\" PRIu64 \"", rc);
+    return RC_SQLITE3_FAILED_FINALIZE;
+  }
+
+  return RC_OK;
+}
+
+retcode_t iota_stor_milestone_exist(const connection_t* const conn,
+                                    const char* index_name,
+                                    const trit_array_p key, bool* exist) {
+  retcode_t ret = RC_OK;
```

Comment:
should assert index_name and maybe rename it align with the "load" query

Reply:
`index_name` can actually be `NULL` in an exist statement

Reply:
forgot that..